gdnative-core/src/core_types/dictionary.rs

```diff
@@ -69,21 +69,45 @@ impl<Access: ThreadAccess> Dictionary<Access> {
         unsafe { (get_api().godot_dictionary_has_all)(self.sys(), keys.sys()) }
     }
 
-    /// Returns a copy of the value corresponding to the key.
+    /// Returns a copy of the value corresponding to the key if it exists.
     #[inline]
-    pub fn get<K>(&self, key: K) -> Variant
+    pub fn get<K>(&self, key: K) -> Option<Variant>
     where
         K: ToVariant + ToVariantEq,
+    {
+        let key = key.to_variant();
+        //This should never return the default Nil, but there isn't a safe way to otherwise check
+        //if the entry exists in a single API call.
+        self.contains(&key).then(|| self.get_or_nil(key))
+    }
+
+    /// Returns a copy of the value corresponding to the key, or `default` if it doesn't exist
+    #[inline]
+    pub fn get_or<K, D>(&self, key: K, default: D) -> Variant
+    where
+        K: ToVariant + ToVariantEq,
+        D: ToVariant + ToVariantEq,
```

Comment:
The default value does not need to be `ToVariantEq`.

Reply:
Oh, I somehow didn't even see the line about relaxing the bound to `OwnedToVariant`. However, it's probably best to track it as a separate issue as you have done anyway, as I'm not sure I fully understand the safety implications and when `ToVariant` will still be required.